src/server/authentication.js

```diff
@@ -87,6 +92,118 @@ module.exports = psql => ({
     }
   },
 
+  /*
+    updateData = {
+      (optional)email, (optional) confirm_email,
+      (optional)username, (optional) confirm_username,
+      (optional)password,  (optional) confirm_password
+    }
+  */
+  // user = { decoded token }
+  updateUserData: async (updateData, user) => {
+    try {
+      const toUpdate = {};
+      const userInDatabase = await psql('users')
+        .where('user_id', '=', user.user_id);
+
+      if (updateData.email) {
+        if (updateData.email.length > EMAIL_MAX) {
```

Comment:
This test could be after-inside the test for if this is a real change

Reply:
If I were to put it inside the test to see if it is a real change, I would need to potentially query the db three times, once for each check: email, username, and password.

The way I have it right now, it just needs to query the db once (drawback being, it queries even if the changes are not real).

Reply:
I had not thought of that. Good call